Client/Frontend/Sync/BraveCore/History/Historyv2.swift

```diff
@@ -0,0 +1,215 @@
+            switch self {
+                case .today:
+                     return Strings.today
+                case .yesterday:
+                     return Strings.yesterday
+                case .lastWeek:
+                     return Strings.lastWeek
+                case .thisMonth:
+                     return Strings.lastMonth
+            }
+        }
+    }
+    
+    // MARK: Lifecycle
+    
+    init(with node: HistoryNode) {
+        self.historyNode = node
+    }
+    
+    // MARK: Internal
+    
+    public var url: String? {
+        historyNode.url.absoluteString
+    }
+    
+    public var title: String? {
+        historyNode.title
+    }
+    
+    public var created: Date? {
+        get {
+            return historyNode.dateAdded
+        }
+        
+        set {
+            historyNode.dateAdded = newValue ?? Date()
+        }
+    }
+    
+    public var domain: String? {
+        historyNode.url.domainURL.absoluteString
+    }
+    
+    public var sectionID: Section? {
+        fetchHistoryTimePeriod(visited: created)
+    }
+    
+    // MARK: Private
+    
+    private let historyNode: HistoryNode
+    private static var observer: HistoryServiceListener?
+    private static let historyAPI = (UIApplication.shared.delegate as? AppDelegate)?.braveCore?.historyAPI
+
+    private func fetchHistoryTimePeriod(visited: Date?) -> Section? {
+        let todayOffset = 0
+        let yesterdayOffset = -1
+        let thisWeekOffset = -7
+        let thisMonthOffset = -31
+        
+        if created?.compare(getDate(todayOffset)) == ComparisonResult.orderedDescending {
+            return .today
+        } else if created?.compare(getDate(yesterdayOffset)) == ComparisonResult.orderedDescending {
+            return .yesterday
+        } else if created?.compare(getDate(thisWeekOffset)) == ComparisonResult.orderedDescending {
+            return .lastWeek
+        } else if created?.compare(getDate(thisMonthOffset))  == ComparisonResult.orderedDescending {
+            return .thisMonth
+        }
+        
+        return nil
+    }
+    
+    private func getDate(_ dayOffset: Int) -> Date {
+        let calendar = Calendar(identifier: Calendar.Identifier.gregorian)
+        let nowComponents = calendar.dateComponents(
+            [Calendar.Component.year, Calendar.Component.month, Calendar.Component.day], from: Date())
+        
+        guard let today = calendar.date(from: nowComponents) else {
+            return Date()
+        }
+        
+        return (calendar as NSCalendar).date(
+            byAdding: NSCalendar.Unit.day, value: dayOffset, to: today, options: []) ?? Date()
+    }
+}
+
+// MARK: History Fetching
+
+extension Historyv2 {
+
+    public class func add(url: URL, title: String, dateAdded: Date, isURLTyped: Bool = true) {
+        guard let historyAPI = Historyv2.historyAPI else {
+            return
+        }
+
+        let historyNode = HistoryNode(url: url, title: title, dateAdded: dateAdded)
```

Comment:
@soner-yuksel - iOS supports special URLs and external URLs as well. `about:` | `mailto:` and `tel:` URLs. Do you know if these URLs are filtered before adding to the sync chain?

Reply:
So the condition of the where the `add` is used is not changed. Only how we add is changed in this context.
History `add` is used inside `func navigateInTab(tab: Tab, to navigation: WKNavigation? = nil) `

And the conditions are 
`if !url.isErrorPageURL, !url.isAboutHomeURL, !url.isFileURL 
`
https://github.com/brave/brave-ios/blob/7391ecf4a767241a9bf0461006b96275b9b3902e/Client/Frontend/Browser/BrowserViewController.swift#L1837

Reply:
This looks good. I think we should add `externalURL` to this list. I can file a follow-up issue for that? Let me know if you disagree.